src/pages/patientView/trialMatch/TrialMatchTable.tsx

```diff
@@ -0,0 +1,354 @@
+                                    </span>
+                                ))}
+                            </div>
+                        </div>
+                        <If condition={index < trial.matches.length - 1}><hr className={styles.criteriaHr}/></If>
+                    </div>
+                ))}
+                </div>
+        },
+        sortBy:(trial: IDetailedTrialMatch) => (trial.matches[0].armDescription),
+        width: this.columnsWidth[ColumnKey.MATCHINGCRITERIA]
+    }, {
+        name: ColumnKey.INTERVENTIONS,
+        render: (trial: IDetailedTrialMatch) => (
+                <ul className={styles.diseasesUl}>
+                    {trial.interventions.map((intervention: any) => (
+                        <li>{intervention}</li>
+                    ))}
+                </ul>
+        ),
+        sortBy:(trial: IDetailedTrialMatch) => (trial.interventions[0]),
+        // download: (trial: IDetailedTrialMatch) => (trial.interventions.join(', ')),
+        width: this.columnsWidth[ColumnKey.INTERVENTIONS]
+    }];
+
+    public tooltipGenomicContent(data: Array<IGenomicGroupMatch>) {
+        const props = this.props;
+        return (
+            <div style={{ maxHeight:400, maxWidth:600, overflow:'auto' }}>
+                {data.map((genomicGroupMatch: any) => (
+                    <div className={styles.genomicSpan}><b>Not </b>{genomicGroupMatch.genomicAlteration.replace(/!/g, '') + ' '}
+                        {genomicGroupMatch.matches[0].sampleIds.map((sampleId: string) => (
+                            <span className={styles.genomicSpan}>
+                                {props.sampleManager!.getComponentForSample(sampleId, 1, '')}
+                            </span>
+                        ))}
+                    </div>
+                ))}
+            </div>
+        );
+    }
+    public tooltipClinicalContent(data: Array<string>) {
+        return (
+            <div style={{ maxHeight:400, maxWidth:600, overflow:'auto' }}>
+                <ul className={styles.alterationUl}>
+                    {data.map((cancerType: string) => (
+                        <li>{cancerType}</li>
+                    ))}
+                </ul>
+            </div>
+        );
+    }
+
+    public mainContent(content: string) {
+        return (
+            <span>
+                <a>{content}</a>
+            </span>
+        );
+    }
+
+    public getHugoSymbolName(matches: Array<IGenomicGroupMatch>, threshold: number) {
+        const hugoSymbolSet = new Set([...matches].map(x => x.genomicAlteration.split(' ')[0].slice(1)));
+        if (hugoSymbolSet.size <= threshold) {
+            return [...hugoSymbolSet].join(', ');
+        }
+        return hugoSymbolSet.size + ' genes' ;
+    }
+
+    public buildDetailedTrialMatches(matches: Array<ITrialMatch>, trials: Array<ITrial>, nctTrials: Array<INctTrial>, sampleManager?: SampleManager) {
```

Comment:
Could just be ITrialMatch[] ...

Reply:
Also why `sampleManager` is optional?

Reply:
PatientViewPage is the father component of TrialMatchTable. I found sample manager could be null in PatientViewPage so I set it's optional in TrialMatchTable.